offchain-modules/packages/x/src/xchain/eth/contract.ts

```diff
@@ -59,12 +61,19 @@ export class EthChain {
     this.provider = new ethers.providers.JsonRpcProvider(url);
     this.bridgeContractAddr = config.contractAddress;
     this.iface = new ethers.utils.Interface(abi);
+    this.asIface = new ethers.utils.Interface(asAbi);
     if (role === 'collector') {
       this.wallet = new ethers.Wallet(config.privateKey, this.provider);
       logger.debug('address', this.wallet.address);
       this.bridge = new ethers.Contract(this.bridgeContractAddr, abi, this.provider).connect(this.wallet);
-      this.assetManager = new ethers.Contract(config.assetManagerContractAddress, assetMangerAbi, this.provider);
+      this.assetManager = new ethers.Contract(config.assetManagerContractAddress, asAbi, this.provider);
       this.multisigMgr = new MultiSigMgr('ETH', this.config.multiSignHosts, this.config.multiSignThreshold);
+
+      // TODO: this line will be removed when publishing.
+      this.assetManagerContract = new ethers.Contract(
+        ethers.utils.computeAddress(ForceBridgeCore.config.eth.privateKey),
+        asAbi,
+      );
```

Comment:
This is because, I use the contract address to choose which abi should be used to parse the log, if the contract is null, then the program will run error.

Reply:
> 

This is because, I use the contract address to choose which abi should be used to parse the log, if the contract is null, then the program will run error.